src/pocketmine/Player.php

```diff
@@ -1356,16 +1356,12 @@ protected function processMovement($tickDiff){
 
 			$this->move($dx, $dy, $dz);
 
-			$diffX = $this->x - $newPos->x;
-			$diffY = $this->y - $newPos->y;
-			$diffZ = $this->z - $newPos->z;
-
 			$yS = 0.5 + $this->ySize;
-			if($diffY >= -$yS or $diffY <= $yS){
-				$diffY = 0;
+			if(-$yS <= $dy or $dy <= $yS){
```

Comment:
So `-$yS <= $dy and $dy <= $yS` can be expressed as `-yS <= dy <= yS` mathematically, which is much more reasonable (`$yS > 0` assumed). Otherwise, this statement is always true.